Frameworks/UIKit/StarboardXaml/StarboardXaml.cpp

```diff
@@ -181,6 +140,52 @@ extern "C" void _ApplicationLaunch(ActivationType activationType, Platform::Obje
     RunApplicationMain(g_principalClassName, g_delegateClassName, startupRect.Width, startupRect.Height, activationType, activationArg);
 }
 
+extern "C" void _ApplicationActivate(Platform::Object^ arguments) {
+    IActivatedEventArgs^ args = static_cast<IActivatedEventArgs^>(arguments);
+    TraceVerbose(TAG, L"OnActivated event received for %d. Previous app state was %d", args->Kind, args->PreviousExecutionState);
+    bool initiateAppLaunch = false;
+    if ((args->PreviousExecutionState != ApplicationExecutionState::Running) &&
+        (args->PreviousExecutionState != ApplicationExecutionState::Suspended)) {
+        TraceVerbose(TAG, L"Initializing application");
+        initiateAppLaunch = true;
+    }
+
+    if (args->Kind == ActivationKind::ToastNotification) {
+        Platform::String^ argsString = safe_cast<ToastNotificationActivatedEventArgs^>(args)->Argument;
+        TraceVerbose(TAG, L"Received toast notification with argument - %s", argsString->Data());
+
+        if (initiateAppLaunch) {
+            _ApplicationLaunch(ActivationTypeToast, argsString);
+        }
+
+        UIApplicationMainHandleToastNotificationEvent(Strings::WideToNarrow(argsString->Data()).c_str());
+    } else if (args->Kind == ActivationKind::VoiceCommand) {
+        Windows::Media::SpeechRecognition::SpeechRecognitionResult^ argResult = safe_cast<VoiceCommandActivatedEventArgs^>(args)->Result;
+        TraceVerbose(TAG, L"Received voice command with argument - %s", argResult->Text->Data());
+
+        if (initiateAppLaunch) {
+            _ApplicationLaunch(ActivationTypeVoiceCommand, argResult);
+        }
+
+        UIApplicationMainHandleVoiceCommandEvent(reinterpret_cast<IInspectable*>(argResult));
+    } else if (args->Kind == ActivationKind::Protocol) {
+        Windows::Foundation::Uri^ argUri = safe_cast<ProtocolActivatedEventArgs^>(args)->Uri;
+        TraceVerbose(TAG, L"Received protocol with uri- %s", argUri->ToString()->Data());
+
```

Comment:
Ideally, we'll use `%hs` and `%ls` to indicate what sort of string we're passing in the args.
Otherwise, `L"%s"` means `L"%ls"` implicitly.